### DISCUSSION TITLE
New Feature: Provide ability to read multi column data as dataframe from a panel or using query just like dashboard variables.

### DISCUSSION THREAD

**Discussion by vellala2000** [grafana/grafana#38211] [Dashboards, Panels & Visualizations]
It would be nice if there is an ability to read multi column data from one dashboard panel or using query just like variables and then use that data to query other data sources.

for example, One panel has table data that is read from mysql db and i want to use that data to query second datasource for the second panel. 

i know Dashboard variables are created for this purpose but unfortunately they hold only one column of data. however in this case i have a table with multiple columns and needs to keep the row intact.

one step further we could even have a repeat option for each entry in a data frame.

**Comment by Gauravshah** [comment]
use mixed panel plugin with transform
or https://github.com/GoshPosh/grafana-meta-queries

**Comment by vellala2000** [comment]
Thank you for your response. 

i did check mixed and meta query link provided. both seems to provide limited functionality. Mixed data source doesn't seems to support feeding output data from one query as input data to another query.

meta quries seems to support data sharing from one query to another how ever it seem to support limited function. 

my use case to get a table of data from one query and use that data to make another query to a different database. i see there are some other people also looking for this feature [](https://stackoverflow.com/questions/63397735/can-we-set-output-of-one-datasource-query-as-input-of-other-datasource-query-in)

**Comment by hugohaggmark** [comment]
This is something that is already on our radar, and we just recently added some "behind the scenes" refactoring to make this scenario easier to do moving forward. But there is still no plan as too when we're moving forward and introduce this as a full-blown feature.

**Comment by vellala2000** [comment]
Thank you. Please prioritize this feature, it would be of great help to avoid duplicates queries and reduce stress on backend data sources..

**Comment by hugohaggmark** [comment]
@vellala2000 the more upvotes the better but I think this issue might be a duplicate, but I don't have the time right now to look into this.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
